extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php

```diff
@@ -395,82 +414,37 @@ public function execute() {
 		 * Make sure we are not running in command line mode where redirects have no sense at all
 		 */
 		if( $this->mIsWikiaActive == 2 && !$this->mCommandLine ) {
-			$this->debug( "city_id={$this->mWikiID};city_public={$this->mIsWikiaActive}), redirected to {$this->mCityHost}" );
+			$this->debug( "city_id={$this->mWikiID};city_public={$this->mIsWikiaActive}), redirected to {$this->mCityUrl}" );
 			header( "X-Redirected-By-WF: 2" );
-			header( "Location: http://{$this->mCityHost}/", true, 301 );
+			header( "Location: {$this->mCityUrl}", true, 301 );
 			wfProfileOut( __METHOD__ );
-			exit(0);
+			return false;
 		}
 
-		/**
-		 * if $this->mCityURL different from city_url we redirect to city_url
-		 * (as main server)
-		 *
-		 * mCityHost may contain path after url (memory-alpha, dofus), we just
-		 * split this for comparing hosts.
-		 */
-		list( $host, $path ) = array_pad( explode( "/", $this->mCityHost, 2 ), 2, false );
+		$url = parse_url( $this->mCityUrl );
 
-		/**
-		 * check if domain from browser is different than main domain for wiki
-		 */
-		$cond1 = !empty( $host ) && !empty( $this->mServerName ) && strtolower( $host ) != $this->mServerName;
+		// check if domain from browser is different than main domain for wiki
+		$cond1 = !empty( $this->mServerName ) &&
+				 ( strtolower( $url['host'] ) != $this->mServerName || $url['path'] !== "/{$this->langCode}" );
 
 		/**
 		 * check if not additional domain was used (then we redirect anyway)
 		 */
-		$cond2 = !empty( $host ) && $this->mAlternativeDomainUsed && ( $host != $this->mOldServerName );
+		$cond2 = $this->mAlternativeDomainUsed && ( $url['host'] != $this->mOldServerName );
 
 		if( ( $cond1 || $cond2 ) && empty( $wgDevelEnvironment ) ) {
-			$url = wfGetCurrentUrl();
-			/**
-			 * now recombine url from parts
-			 */
-			if( preg_match( "!^/$path!", $url[ "path" ] ) == 0 ) {
-				$url[ "path" ] = "/{$path}" . $url[ "path" ];
-			}
-
-			// Special treatment for paragon.wiki
-			// For URLs starting with upper-case letter (/Something) redirect to /wiki/Something
-			$wikiPrefix = '';
-			if ( $host == 'paragon.wikia.com' ) {
-				$firstLetter = substr( $url['path'], 1, 1 );
-				if ( is_string( $firstLetter ) && ctype_upper( $firstLetter ) ) {
-					$wikiPrefix = '/wiki';
-				}
-			}
+			$url = rtrim( $this->mCityUrl, '/' );
```

Comment:
Minor suggestion: Since `$url` before and after this block is a parsed URL array, I'd avoid overwriting it to a simple string in this block just for code clarity (for example, while reviewing the part below setting the script path, I looked back up for the last reference to `$url` and found this, and needed to double check it would be an array then). We could probably even just have `$target = rtrim( $this->mCityUrl, '/' ) . '/' . $this->pathParams;` below. What do you think?

Reply:
good point, this was not intentional :)